crawl-ref/source/ouch.cc

```diff
@@ -807,16 +807,20 @@ int do_shave_damage(int dam)
 }
 #endif
 
-// Determine what's threatening for purposes of sacrifice drink and reading.
-// the statuses are guaranteed not to happen if the incoming damage is less
-// than 5% max hp. Otherwise, they scale up with damage taken and with lower
-// health, becoming certain at 20% max health damage.
-static bool _is_damage_threatening (int damage_fraction_of_hp)
+// Determine what's threatening for purposes of sacrifice drink and no scroll
+// mutation. The statuses are guaranteed not to happen if the incoming damage
+// is less than 15/10/5% max hp based on the mutation tier. Otherwise, they 
+// scale up with damage taken and with lower health, becoming certain at 
+// 60/40/20% max health damage also based on the mutation tier.
+static bool _is_damage_threatening (int damage_fraction_of_hp, int mut_level = 3)
 {
+    const int safety_level = 4 - mut_level;
     const int hp_fraction = you.hp * 100 / you.hp_max;
-    return damage_fraction_of_hp > 5
-            && hp_fraction <= 85
-            && (damage_fraction_of_hp + random2(20) >= 20
+    const int safe_damage_fraction = 5 * safety_level;
+    const int scary_damage_fraction = 20 * safety_level;
+    return damage_fraction_of_hp > safe_damage_fraction
+            && hp_fraction <= 100 - scary_damage_fraction + safe_damage_fraction
+            && (damage_fraction_of_hp + random2(20) >= scary_damage_fraction
                 || random2(100) > hp_fraction);
```

Comment:
The random2 part of this formula doesn't match the behavior I would expect. damage_fraction_of_hp + random2(20) >20 can pass for any damage fraction, but this can't occur for scary_damage_fraction of 40 or 60; you effectively have a higher safety threshold on that part of the argument bc you need a damage fraction of 21 or 41 to have any chance to pass it (can still pass the other random2 on hp_fraction of course). The random2 should probably be random2(20* mut_level) instead.